webpack/assets/javascripts/application.js

```diff
@@ -0,0 +1,7 @@
+require('expose?$!expose?jQuery!jquery');
```

Comment:
No, this is so we expose the `$` and `jQuery` variable in the global scope (for the old code to work). 
IIUC, the line in the webpack config has to do with which modules are available to the webpack managed code.